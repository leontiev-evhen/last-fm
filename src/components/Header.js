import React from 'react';

export function Header() {
  return (
    <header>
      <div  className="header-section section-title">
        <h2>Last fm</h2>
      </div>
    </header>
  );
}
