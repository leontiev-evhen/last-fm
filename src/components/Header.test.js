import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';

describe('Header component', () => {
  const component = shallow(<Header />);
  it('render correctly Header component', () => {
    expect(component).toMatchSnapshot();
  });
  it('render correctly Header component and Link', () => {
    expect(component.find('Link')).toHaveLength(1);
  });
});