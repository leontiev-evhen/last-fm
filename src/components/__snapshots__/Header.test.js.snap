// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component render correctly Header component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="header-section section-title"
      >
        <h2>
          <Link
            replace={false}
            to="/"
          >
            Last fm
          </Link>
        </h2>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h2>
          <Link
            replace={false}
            to="/"
          >
            Last fm
          </Link>
        </h2>,
        "className": "header-section section-title",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/"
          >
            Last fm
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Last fm",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Last fm",
          "type": [Function],
        },
        "type": "h2",
      },
      "type": "div",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="header-section section-title"
        >
          <h2>
            <Link
              replace={false}
              to="/"
            >
              Last fm
            </Link>
          </h2>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2>
            <Link
              replace={false}
              to="/"
            >
              Last fm
            </Link>
          </h2>,
          "className": "header-section section-title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/"
            >
              Last fm
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Last fm",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Last fm",
            "type": [Function],
          },
          "type": "h2",
        },
        "type": "div",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
